osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs

```diff
@@ -40,6 +40,17 @@ public abstract class RoomSubScreen : OnlinePlaySubScreen, IPreviewTrackOwner
 
         private IBindable<WeakReference<BeatmapSetInfo>> managerUpdated;
 
+        [Cached]
+        protected readonly MultiplayerBeatmapAvailablilityTracker BeatmapAvailablilityTracker;
+
+        protected RoomSubScreen()
+        {
+            InternalChild = BeatmapAvailablilityTracker = new MultiplayerBeatmapAvailablilityTracker
```

Comment:
I really don't like this. Feels like it could go wrong at any point (overwriting other children etc). Use `AddInternal` instead. And probably move to BDL below.

Reply:
right, was thinking this will be the top constructor, but that's actually bad practice regardless